app/upgrades/v8/upgrades.go

```diff
@@ -49,6 +52,65 @@ func FixBankMetadata(ctx sdk.Context, keepers *keepers.AppKeepers) error {
 	return nil
 }
 
+func QuicksilverFix(ctx sdk.Context, keepers *keepers.AppKeepers) error {
+	// Refund stuck coins from ica address
+	sourceAddress, err := sdk.AccAddressFromBech32("cosmos13dqvh4qtg4gzczuktgnw8gc2ewnwmhdwnctekxctyr4azz4dcyysecgq7e")
+
+	if err != nil {
+		return errors.New("invalid source address")
+	}
+	destinationAddress, err := sdk.AccAddressFromBech32("cosmos1jc24kwznud9m3mwqmcz3xw33ndjuufnghstaag")
+	if err != nil {
+		return errors.New("invalid destination address")
+	}
+
+	// Get balance from stuck address and subtract 1 uatom sent by bad actor
+	sourceBalance := keepers.BankKeeper.GetBalance(ctx, sourceAddress, "uatom")
+	refundBalance := sourceBalance.SubAmount(sdk.NewInt(1))
+	keepers.BankKeeper.SendCoins(ctx, sourceAddress, destinationAddress, sdk.NewCoins(refundBalance))
+
+	// Get connection to quicksilver chain
+	connectionId, err := getConnectionIdForChainId(ctx, keepers, "quicksilver-1")
+	if err != nil {
+		return err
+	}
+
+	// Close channels
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.deposit")
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.withdrawal")
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.performance")
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.delegate")
+
+	return nil
+}
+
+func closeChannel(keepers *keepers.AppKeepers, ctx sdk.Context, connectionId string, port string) {
+	activeChannelId, found := keepers.ICAHostKeeper.GetActiveChannelID(ctx, connectionId, port)
+	if found {
+		channel, found := keepers.IBCKeeper.ChannelKeeper.GetChannel(ctx, icatypes.PortID, activeChannelId)
```

Comment:
Maybe an extra check you could add here is to make sure that the channel is indeed in state `OPEN`.

Reply:
What happens here with `icacontroller-cosmoshub-4.delegate` - it is currently in `TRYOPEN`. Presumably it won't be `CLOSED` by this. Do we need it to be to be able to reopen @crodriguezvega ?

Reply:
Oh - we have both 465 and 466 that are both `icacontroller-cosmoshub-4.delegate`. I guess this is okay?

Reply:
Yes, I think it is fine the way it is because [`channel-465` is `OPEN` state](https://rest.cosmos.directory/cosmoshub/ibc/core/channel/v1/channels/channel-465/ports/icahost) and `channel-466` is in `TRYOPEN` state](https://rest.cosmos.directory/cosmoshub/ibc/core/channel/v1/channels/channel-466/ports/icahost), but 465 is the one that got stored as active channel in state because that's the channel that completed the handshake (storing the active channel happens in the `OnChanOpenConfirm` callback](https://github.com/cosmos/ibc-go/blob/v3.0.0/modules/apps/27-interchain-accounts/host/keeper/handshake.go#L102)). Related to the [comment above](https://github.com/cosmos/gaia/pull/1976#discussion_r1059655284), maybe we could use the specific channel IDs that we want to close, to be extra sure.

Reply:
Should we also close channel-466?

Reply:
@AdityaSripal

Reply:
Hi @damiannolan, if you or someone in ibc team can do a final review of this ? We plan to include it in the next release if possible, this week. 

Thank you so much !